pkg/syncer/util.go

```diff
@@ -874,38 +873,25 @@ func getPatchData(oldObj, newObj interface{}) ([]byte, error) {
 
 // PatchStoragePolicyUsage patches the StoragePolicyUsage CR based on old and new objects
 func PatchStoragePolicyUsage(ctx context.Context, cnsOperatorClient client.Client,
-	oldObj *storagepolicyusagev1alpha1.StoragePolicyUsage,
-	newObj *storagepolicyusagev1alpha1.StoragePolicyUsage) error {
+	newObj *storagepolicyusagev1alpha2.StoragePolicyUsage) (err error) {
 	log := logger.GetLogger(ctx)
-	patch, err := getPatchData(oldObj, newObj)
-	if err != nil {
-		log.Errorf("error fetching PatchData StoragePolicyUsage CR. err: %v", err)
-		return err
-	}
-	patch, err = addResourceVersion(patch, oldObj.ResourceVersion)
-	if err != nil {
-		log.Errorf("applying ResourceVersion to patch data failed: %v", err)
-		return err
-	}
-	rawPatch := client.RawPatch(apitypes.MergePatchType, patch)
-
 	// Try to patch StoragePolicyUsage CR for allowedRetries times
 	allowedRetries := allowedRetriesToPatchStoragePolicyUsage
 	attempt := 0
 	for {
 		attempt++
-		err = cnsOperatorClient.Patch(ctx, oldObj, rawPatch)
+		err = cnsOperatorClient.Status().Update(ctx, newObj)
```

Comment:
check if we need to continue using Patch instead of Update ?

Reply:
Always patch, updates are flaky.

Reply:
did change to Status().Patch